test/HooksUseReducerTest.re

```diff
@@ -0,0 +1,245 @@
+
+    updateContainer(container, <ComponentWithState />);
+
+    validateStructure(rootNode, expectedStructure);
+  });
+
+  module ComponentThatDispatchesIncreaseAction = (
+    val component(
+          (render, ~children, ~event: Event.t(unit), ~initialValue: int, ()) =>
+          render(
+            () => {
+              /* Hooks */
+              let (s, dispatch) = useReducer(reducer, initialValue);
+              /* End hooks */
+
+              useEffect(() => {
+                let unsubscribe =
+                  Event.subscribe(event, () => dispatch(Increase));
+                () => unsubscribe();
+              });
+
+              <aComponent testVal=s />;
+            },
+            ~children,
+          )
+        )
+  );
+
+  test("useReducer updates state with dispatch function", () => {
+    let rootNode = createRootNode();
+
+    let container = createContainer(rootNode);
+
+    let event: Event.t(unit) = Event.create();
+
+    updateContainer(
+      container,
+      <ComponentThatDispatchesIncreaseAction event initialValue=2 />,
+    );
+
+    Event.dispatch(event, ());
+
+    let expectedStructure: tree(primitives) =
+      TreeNode(Root, [TreeLeaf(A(3))]);
+    validateStructure(rootNode, expectedStructure);
+  });
+
+  test("useReducer doesn't leak state between components", () => {
+    let rootNode = createRootNode();
+
+    let container = createContainer(rootNode);
+
+    let event: Event.t(unit) = Event.create();
+
+    updateContainer(
+      container,
+      <ComponentThatDispatchesIncreaseAction event initialValue=2 />,
+    );
+
+    Event.dispatch(event, ());
+
+    let expectedStructure: tree(primitives) =
+      TreeNode(Root, [TreeLeaf(A(3))]);
+    validateStructure(rootNode, expectedStructure);
+
+    updateContainer(container, <ComponentWithState />);
+
+    /* The 'componentWithState' should have its own state, so it should revert back to 2 - */
+    /* and not pick up the state from the previous component */
+    let expectedStructure: tree(primitives) =
+      TreeNode(Root, [TreeLeaf(A(2))]);
+    validateStructure(rootNode, expectedStructure);
+  });
+
+  test("useReducer dispatch state persists across renders", () => {
+    let rootNode = createRootNode();
+
+    let container = createContainer(rootNode);
+
+    let event: Event.t(unit) = Event.create();
+
+    updateContainer(
+      container,
+      <ComponentThatDispatchesIncreaseAction event initialValue=2 />,
+    );
+
+    Event.dispatch(event, ());
+
+    let expectedStructure: tree(primitives) =
+      TreeNode(Root, [TreeLeaf(A(3))]);
+    validateStructure(rootNode, expectedStructure);
+
+    updateContainer(
+      container,
+      <ComponentThatDispatchesIncreaseAction event initialValue=2 />,
+    );
+
+    let expectedStructure: tree(primitives) =
+      TreeNode(Root, [TreeLeaf(A(3))]);
+    validateStructure(rootNode, expectedStructure);
+  });
+
+  test("useReducer can update multiple times", () => {
```

Comment:
Excellent test coverage @jchavarri - thanks for exercising the new functionality so thoroughly! 💯